public/app/plugins/datasource/prometheus/configuration/ConfigEditor.test.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import { FieldValidationMessage } from '@grafana/ui';
+
+import { validateInput } from './ConfigEditor';
+import { DURATION_REGEX } from './PromSettings';
+
+// replaces promSettingsValidationEvents to display a <FieldValidationMessage> onBlur for duration input errors
+describe('promSettings validateInput', () => {
+  it.each`
+    value    | expected
+    ${'1ms'} | ${true}
+    ${'1M'}  | ${true}
+    ${'1w'}  | ${true}
+    ${'1d'}  | ${true}
+    ${'1h'}  | ${true}
+    ${'1m'}  | ${true}
+    ${'1s'}  | ${true}
+    ${'1y'}  | ${true}
```

Comment:
will multiple duration types work? i.e `1h 5m 2s`?

Reply:
Good catch! No it does not, this was a custom prom settings duration validation
```
export const promSettingsValidationEvents = {
  [EventsWithValidation.onBlur]: [
    regexValidation(
      /^$|^\d+(ms|[Mwdhmsy])$/,
      'Value is not valid, you can use number with time unit specifier: y, M, w, d, h, m, s'
    ),
  ],
};
```

This is different from the `isValidDuration` duration check that you imported for incremental querying
```
/**
 * isValidDuration returns true if the given string can be parsed into a valid Duration object, false otherwise
 *
 * @param durationString - string representation of a duration
 *
 * @public
 */
export function isValidDuration(durationString: string): boolean {
  for (const value of durationString.trim().split(' ')) {
    const match = value.match(/(\d+)(.+)/);
    if (match === null || match.length !== 3) {
      return false;
    }

    const key = Object.entries(durationMap).find(([_, abbreviations]) => abbreviations?.includes(match[2]))?.[0];
    if (!key) {
      return false;
    }
  }

  return true;
}
```

For config items 'Query timeout' and 'Scrape interval' we never supported multiple durations. For incremental querying, do we support multiple durations?

Reply:
Yeah but just because that function does, if this is the desired functionality then we should be good

Reply:
updated validation for incremental querying! https://github.com/grafana/grafana/pull/66198/commits/b9a0f01e3c9b25a1ac39c33b53a1d5d9f419e350